api/src/endpoints/getNotificationsEndpoint.ts

```diff
@@ -0,0 +1,174 @@
+import express from 'express';
+import { CommentId, UserId, Comment } from 'entities';
+import { Gateways, Result, err } from 'gateways';
+import { getAuthenticatedUserId } from '../parseAuth0User';
+import { authenticationRequiredError } from 'gateways/src/errors';
+import { VizId } from '@vizhub/viz-types';
+import { VizNotificationRequestResult } from 'entities/src/Notifications';
+
+//Everything the UI needs to display the notifications UI is returned from this endpoint
+export const getNotificationsEndpoint = ({
+  app,
+  gateways,
+}: {
+  app: any;
+  gateways: Gateways;
+}) => {
+  const {
+    getNotificationsByUserId,
+    getComment,
+    getInfo,
+    getUser,
+  } = gateways;
+  app.post(
+    '/api/get-notifications',
+    express.json(),
+    async (req, res) => {
+      if (req.body) {
+        const {
+          userId,
+        }: {
+          userId: UserId;
+        } = req.body;
+
+        // Need to be authenticated to get notifications.
+        const authenticatedUserId =
+          getAuthenticatedUserId(req);
+        if (!authenticatedUserId) {
+          res.json(err(authenticationRequiredError()));
+          return;
+        }
+
+        // Validate relevant user is logged in
+        if (userId !== authenticatedUserId) {
+          res.json(err(authenticationRequiredError()));
+          return;
+        }
+
+        console.log('pre notif get');
+
+        const getNotificationsByUserIdResult =
+          await getNotificationsByUserId(userId);
+
+        if (
+          getNotificationsByUserIdResult.outcome ===
+          'failure'
+        ) {
+          res.send(getNotificationsByUserIdResult);
+          return;
+        }
+
+        console.log('Starting Notif get');
+
+        const notifications =
+          getNotificationsByUserIdResult.value
+            .map((snapshot) => snapshot.data)
+            .filter((notification) => !notification.read);
+
+        const commentsPromises = notifications.map(
+          async (notification) =>
+            await getComment(notification.commentId),
```

Comment:
Ideal would be a single DB read:
```
            await getComments(notifications.map(notification => notification.commentId)),
```

but that would require some custom development of this new function `getComments`